### ISSUE TITLE
Cloud Formatter incorrectly Escaping JSON

### ISSUE THREAD

**Issue by prefectcboyd** [PrefectHQ/prefect-ui-library#1555]
Attempting to pass in a very simple custom parameter block, where I'm typing the following literally:

```
{
"test": true
}
```

What occurs instead is:
```
"\"\\\"\\\\\\\"\\\\\\\\\\\\\\\"\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\"\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\"{\\\\\\\\\\\\\\\"t\\\\\\\"e\\\"s\"t"
```

These don't match 100% because it becomes a little hard to follow what I even typed already, but this occurs elsewhere where input is processed, such as Environment Variables in work-pools and I'm sure other locations.

Screenshots attached showcasing the behavior - this can be reproduced by attempting to type / format in any JSON fields.
![image (17)](https://github.com/PrefectHQ/prefect-ui-library/assets/105737359/e1f04457-cc33-4f49-8e6c-b6c166f7ea2d)
![image (16)](https://github.com/PrefectHQ/prefect-ui-library/assets/105737359/791343ca-1303-448e-aac6-263b00703b53)
![image (15)](https://github.com/PrefectHQ/prefect-ui-library/assets/105737359/bf7c2560-26ae-4632-b70d-3a6dbc0f5fba)

**Comment by jbbrunello** [comment]
The multiple escape gibberish is gone, good riddance. However there is still undue formatting going on: for example, double quotes are being removed **for no reason**. If I type in `"12345"` (with double quotes) the Cloud Formatter replaces my input with `12345` without double quotes. If I put escaped quotes around my string such as `"\"12345\""` the Cloud Formatter still replaces my input with `12345` (removig both the enclosing **and the escaped double quotes**). **There is no reason for this**: if I need to pass to my flow a string parameter (and if the string contains only numeric characters that's my business: it's still a valid string and valid JSON) I need to be able to do just this regardless of the content of the string. **More generally, there is no need to aggressively reformat inputs** (unless I optionally click on the "Format" button). Please fix this asap as it's a **blocking issue**.

**Comment by jbbrunello** [comment]
I confirm this is a blocking issue: if I input legal string (such as `"True"`) the Cloud Formatter removes the double quotes **then complains it's not valid JSON and inhibits the "submit" button**, preventing me from launching any flow run that requires parameters.
![image](https://github.com/PrefectHQ/prefect-ui-library/assets/138983357/b19c03ae-7a11-40e5-a731-8aa488407756)

**Comment by jbbrunello** [comment]
Moreover, even switching to direct JSON formatting doesn't solve the issue: in this example (with a perfeclty valid JSON) clicking on Submit yields no effect: **the flow run does not start**.
![image](https://github.com/PrefectHQ/prefect-ui-library/assets/138983357/4e1e1e40-56cb-42b6-9a7d-5843d77d9f92)
